src/app/icd/client/ICDRefreshKeyInfo.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "ICDRefreshKeyInfo.h"
+#include "CheckInDelegate.h"
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/CommandPathParams.h>
+#include <app/InteractionModelEngine.h>
+#include <app/InteractionModelTimeout.h>
+#include <app/OperationalSessionSetup.h>
+
+namespace chip {
+namespace app {
+
+ICDRefreshKeyInfo::ICDRefreshKeyInfo(CheckInDelegate * apCheckInDelegate, const ICDClientInfo aICDClientInfo,
```

Comment:
Why are we passing the clientinfo by value?

Reply:
ICDClientInfo sent from CheckInHandler to the OnRefreshKeyNeeded delegate is a local variable within OnMessageReceived. So, oassing the clientInfo by reference for creating ICDRefreshKeyInfo.